File: VB/13_Conversion/ToLinearizedPdf.vb
Imports Spire.Pdf.Conversion

Namespace ToLinearizedPdf
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			' Specify the file path of the input PDF file
			Dim input As String = "../../../../../../Data/PDFTemplate_N.pdf"

			' Specify the file path for the resulting linearized PDF file
			Dim result As String = "toLinearizedPdf.pdf"

			' Create a new instance of PdfToLinearizedPdfConverter
			Dim converter As New PdfToLinearizedPdfConverter(input)

			' Convert the input PDF to a linearized PDF and save it
			converter.ToLinearizedPdf(result)

			' Dispose the convertor
			converter.Dispose()

			' Launch the Pdf file
			PDFDocumentViewer(result)
		End Sub

		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS/13_Conversion/ToLinearizedPdf.cs
using System;
using System.Windows.Forms;
using Spire.Pdf.Conversion;


namespace ToLinearizedPdf
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            // Specify the path of the input PDF file.
            string input = "../../../../../../Data/PDFTemplate_N.pdf";

            // Specify the desired file name for the resulting linearized PDF.
            string result = "toLinearizedPdf.pdf";

            // Create an instance of the PdfToLinearizedPdfConverter class with the input PDF file path.
            PdfToLinearizedPdfConverter converter = new PdfToLinearizedPdfConverter(input);

            // Convert the input PDF file to a linearized PDF and save it using the specified result file name.
            converter.ToLinearizedPdf(result);

            //Launch the Pdf file
            PDFDocumentViewer(result);
        }

        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}
